src/client/encrypt.js

```diff
@@ -42,13 +42,31 @@ module.exports = function (client, options) {
       }
 
       function sendEncryptionKeyResponse () {
+        const mcData = require('minecraft-data')(client.version)
+
         const pubKey = mcPubKeyToPem(packet.publicKey)
         const encryptedSharedSecretBuffer = crypto.publicEncrypt({ key: pubKey, padding: crypto.constants.RSA_PKCS1_PADDING }, sharedSecret)
         const encryptedVerifyTokenBuffer = crypto.publicEncrypt({ key: pubKey, padding: crypto.constants.RSA_PKCS1_PADDING }, packet.verifyToken)
-        client.write('encryption_begin', {
-          sharedSecret: encryptedSharedSecretBuffer,
-          verifyToken: encryptedVerifyTokenBuffer
-        })
+
+        if (mcData.supportFeature('signatureEncryption')) {
+          // 1.19+
```

Comment:
I think specifying the version range next to each support feature call is not necessary